plugins/thunderbolt/fu-plugin-thunderbolt.c

```diff
@@ -183,20 +185,42 @@ fu_plugin_thunderbolt_add (FuPlugin *plugin, GUdevDevice *device)
 		return;
 	}
 
+	is_host = fu_plugin_thunderbolt_is_host (device);
+
 	vid = fu_plugin_thunderbolt_udev_get_id (device, "vendor", &error);
 	if (vid == 0x0) {
 		g_warning ("failed to get Vendor ID: %s", error->message);
-		return;
+		if (is_host)
+			is_safemode = TRUE;
```

Comment:
Yeah, I think that makes sense. We should also change the check for interesting devies (line 164/170) to to see if we have `nvm_version`. And if reading the value is failing with `ENODATA` we set `is_safe_mode` do the safe mode special casing. We probably also want to generate a special unique_id, assuming we dont have that anymore in safe mode either.